AutoFilterer.Dynamics/DynamicFilter.cs

```diff
@@ -0,0 +1,137 @@
+    {
+        public string Value { get; set; }
+        public static implicit operator DynamicFilter(string value) => new DynamicFilter(value);
+        public static explicit operator string(DynamicFilter value) => value.Value;
+
+        private static readonly Dictionary<string, IFilterableType> specialKeywords = new Dictionary<string, IFilterableType>
+        {
+            { "eq", new OperatorComparisonAttribute(OperatorType.Equal) },
+            { "not", new OperatorComparisonAttribute(OperatorType.NotEqual) },
+            { "gt", new OperatorComparisonAttribute(OperatorType.GreaterThan) },
+            { "gte", new OperatorComparisonAttribute(OperatorType.GreaterThanOrEqual) },
+            { "lt", new OperatorComparisonAttribute(OperatorType.LessThan) },
+            { "lte", new OperatorComparisonAttribute(OperatorType.LessThanOrEqual) },
+            { "contains", new StringFilterOptionsAttribute(StringFilterOption.Contains) },
+            { "endswith", new StringFilterOptionsAttribute(StringFilterOption.EndsWith) },
+            { "startswith", new StringFilterOptionsAttribute(StringFilterOption.StartsWith) },
+        };
+
+        public CombineType CombineWith { get; set; }
+
+        public DynamicFilter()
+        {
+        }
+
+        public DynamicFilter(string value)
+        {
+            this.Value = value;
+        }
+
+        public override string ToString() => this.Value;
+
+        public IQueryable<TEntity> ApplyFilterTo<TEntity>(IQueryable<TEntity> query)
+        {
+            var parameter = Expression.Parameter(typeof(TEntity), "x");
+
+            var exp = BuildExpression(typeof(TEntity), parameter);
+            if (exp == null)
+                return query;
+
+            if (exp is MemberExpression || exp is ParameterExpression)
+                return query;
+
+            var lambda = Expression.Lambda<Func<TEntity, bool>>(exp, parameter);
+            return query.Where(lambda);
+        }
+
+        public Expression BuildExpression(Type entityType, Expression body)
+        {
+            Expression finalExpression = body;
+
+            foreach (var key in this.Keys)
+            {
+                var filterValue = new DynamicFilter(this[key]);
+                if (IsPrimitive(key))
+                {
+                    var targetProperty = entityType.GetProperty(key);
+                    var value = Convert.ChangeType((string)filterValue, targetProperty.PropertyType);
+                    var exp = new OperatorComparisonAttribute(OperatorType.Equal).BuildExpression(body, targetProperty, filterProperty: null, value);
+
+                    var combined = Combine(finalExpression, exp);
+                    finalExpression = Combine(body, combined);
+                }
+                else
+                {
+                    var splitted = key.Split('.');
+                    if (splitted.Length == 2)
+                    {
```

Comment:
Can be recursive function? @enisn

Reply:
This is a temporary solution until inner objects implemented.
I've just made that check as method to better understanding what does it check.

You can check Commit 000c8328